### ISSUE TITLE
challenges getbytag returns filter tag as a string

### ISSUE THREAD

**Issue by JMurphyWeb** [emfoundation/ce100-app#467]
challenges getbytag returns filter tag as a string, but should return as an object, so that we have the name and the id of the tag.

here is the desired shape of the filter_tag key in order to bring it in line with the equivalent organisations query.
```
filter: { id: 52, name: 'Access over ownership' }
```
If there is no tag, it should return as null as opposed to undefined

**Comment by JMurphyWeb** [comment]
Here is the result of the bug:

 in browse / explore view
- when filtering with a tag, often the 'All Tags' doesn't change to the filter tag's name
- when filtering with a tag, clicking on challenges/organisations removes the search criteria when it should preserve the filter tag

**Comment by JMurphyWeb** [comment]
I have fixed the returning filter and made a pr in the challenges plugin.

But there is a new issue:
When there are no results from a filtered browse we should still get the name of the filter tag back from the query.  

In orgs, we do return the filter:
![screen shot 2016-12-14 at 12 07 34](https://cloud.githubusercontent.com/assets/12638261/21181524/45b9655a-c1f6-11e6-87fa-8841add4f83c.png)

In challenges, we do not:
![screen shot 2016-12-14 at 12 07 42](https://cloud.githubusercontent.com/assets/12638261/21181527/49e17a50-c1f6-11e6-9234-f280d7236c8e.png)

**Comment by JMurphyWeb** [comment]
I will compare the two queries.

**Comment by JMurphyWeb** [comment]
the key seemed to be using outer joins after the join with the tags table, and unsuring we used IS NOT FALSE as opposed to TRUE, and finally including and OR `tags.id = tagId`

```
    OR tags.id = 1
    AND tags.active IS NOT FALSE 
    AND challenges.active IS NOT FALSE 
    AND organisations.active IS NOT FALSE 
    challenges.title ASC;
```

**Comment by SimonLab** [comment]
resolved with #470
